### ISSUE TITLE
All scenarios end up in latest feature in report

### ISSUE THREAD

**Issue by vrnvorona** [jonsamwell/flutter_gherkin#128]
Today i noticed that in my reports all scenarios are not separated by feature files, and are instead all bunched in latest one.

For me this happens because auto generated file's gherkin_suite_test.g.dart function onRun() is executed as if parallel:

```
void onRun() {
    print('---<debug123>---');
    testFeature0();
    print('---<debug123>---');
    testFeature1();
    print('---<debug123>---');
    testFeature2();
  }
```

All those prints will be instant before first scenario is running. This behaviour is constant until i hit testWidgets part of code, then things are running consecutively.

```
void runScenario(
    String name,
    Iterable<String> tags,
    Future<void> Function(TestDependencies dependencies) runTest,
  ) {
    if (_evaluateTagFilterExpression(configuration.tagExpression, tags)) {
      print('---<debugscen-1>---');
      testWidgets(
        name,
        (WidgetTester tester) async {
          print('---<debugscen0>---');
          // rest of code
})}}
```

-1s will be printed instantly, and 0s will be printed as scenarios run.

Sorry for such bad debugging, but what i have is all. I tried figuring out how to fix it, but nothing worked.

**Comment by vrnvorona** [comment]
https://codebeautify.org/jsonviewer/cbc82ecf example report with such behaviour. Ironically that last feature file is empty

**Comment by jonsamwell** [comment]
@vrnvorona thanks for reporting this.  Yes it has to run syncronisly first to register all the `test` steps although the report should still be correct.  I'll take a look.

What version are you using the lastest 3-rc1?

**Comment by vrnvorona** [comment]
@jonsamwell 1.2.0 flutter_gherkin and 1.1.10 dart_gherkin. At least judging by pubspec.lock

**Comment by jonsamwell** [comment]
Could you upgrade to the latest (https://pub.dev/packages/flutter_gherkin/versions/3.0.0-rc.1) as you are using the new integration_test way of doing things and see if you still get this error.

**Comment by vrnvorona** [comment]
I will try with sample app or something, since our project is not yet on Flutter 2 sadly.

**Comment by jonsamwell** [comment]
Ah no worries.  I don't blame you not upgrading to Flutter 2 yet.  I have been migrating a medium sized app all week to Flutter 2 and null-safety it is a HUGE pain

**Comment by umangdivyanshu** [comment]
Hi Jon,
I am also facing the same issue and cucumber-html-reporter is not able to generate html report from the integration_response_data.json. Waiting for this issue to be fixed. :)

**Comment by jonsamwell** [comment]
Fixed in 3.0.0-rc.3

See https://github.com/jonsamwell/flutter_gherkin/blob/3.0.0-rc.3/example_with_integration_test/generate-cucumber-html-report.js on how to generate html report from output
